test suites: replace ?t: with test_server:

max-au opened the pull request:
Removing `?t` previously retained for backward compatibility allows to
search for test_server callsites easier.

u3s commented on lib/snmp/test/exp/snmp_agent_bl_test.erl:
```diff
@@ -4993,14 +4993,14 @@ otp_4394_config(AgentDir, MgrDir, Ip0) ->
     ?line update_vacm(Vsn, AgentDir),
     Ta1 = {"shelob v1", 
-	   [134,138,177,177], 5000, 1500, 3, %% Anv�nd Ip och modda
+	   [134,138,177,177], 5000, 1500, 3, %% Anv�nd Ip och modda
```
Do you know what happened with swedish character here?

KennethL replied:
If swedish is used in a comment please translate it to english at least if the line with the comment is touched in a commit.
We should not have swedish in the comments.

max-au replied:
It was unintended, I will drop accidental changes caused by improper editor that broke non-ASCII characters. I think it might be better to clean up comments in a separate patch for review convenience.

u3s commented on lib/stdlib/test/gen_statem_SUITE.erl:
```diff
@@ -2368,7 +2368,7 @@ init(stop) ->
     {stop,shutdown};
 init(sleep) ->
-    ?t:sleep(1000),
+    ct:sleep(1000),
```
is replacement with ct and not test_server intended? functions has a different API and also calling ct:sleep will invoke test_server:adjusted_sleep providing other logic than test_server:sleep...

max-au replied:
Yes, I'd like to suggest this change. If my understanding it correct, this test suite needs "at least X" milliseconds, so adjusting that X should not hurt.
BTW are there official guidelines which method to use, `receive after 1000 -> ok end`, or `timer:sleep`, or `ct:sleep`, or `test_server:sleep`?

IngelaAndin replied:
test_server:sleep is legacy, e.i it is from the original internal tool that now only exists as an internal part of the official common_test application. So it should not be used. I would think ct:sleep is best in the cases that there is no better alternative to sleep in a test suite as it adopts the sleep period for slower test runs such as when you run cover for instance.

u3s commented on lib/common_test/include/ct.hrl:
```diff
@@ -39,5 +39,4 @@
 %% Backward compatibility for test_server test suites.
 %% DO NOT USE IN NEW TEST SUITES.
 -define(line,).
--define(t,test_server).
```
such definition is copy/pasted(?) in several other files (e.g. disk_log_SUITE.erl) and seems unused. can you remove them as well in this PR?

IngelaAndin commented:
@max-au I think it looks good. However, I am thinking this is a test suite refresh, has no impact on functionality but changes a lot of test code, perhaps we ought to rebase it on maint to reduce the maintenance burden.  Also when you are at it maybe change  ?t:fail to ct:fail for the same reason as sleep.

max-au commented:
Cleaning spree. 
As suggested, rebased to maint, and replaced all (except those in test_server_SUITE itself) invocations of `test_server:fail` with corresponding `ct:fail` calls.

IngelaAndin commented:
Thank you for your work on this.
